docs/TuxActionStores.md

```diff
@@ -0,0 +1,179 @@
+
+```
+    var actionStore = ActionStores.createStore(storeProps);
+```
+
+#### <a id="storeProps"></a>Parameter - `storeProps` - type: OBJECT - required [#](#storeProps)
+StoreProps is an object containing actions that will encapsulate the store's data, with an optional register method which, if provided, will automatically register all the actions with the newly created store. Let's see what this looks like:
+
+```
+    var messageStoreProperties = {
+      _messages: [],
+
+      onCreate: function (message) {
+        this._messages.push(message);
+        this.emitChange();
+      },
+
+      register: function () {
+        return {
+          messages: {
+            create: this.onCreate
+          }
+        };
+      }
+    };
+
+```
+
+##### <a id="storeProps-register"></a>Method - `storeProps.register` - type: FUNCTION - optional [#](#createActionStore)
+The `register` method is an optional key on the `storeProps` parameter which, if defined, will invoke `Actions.register`, passing in the store as the first argument and the object returned by this `storeProps.register` method as the second argument. Here is what the `register` method performs on the backend:
+
+```
+    Actions.register(actionStore, actionStore.register.call(actionStore));
+```
+
+ Now lets look at an example of `storeProps.register`:
+
+```
+    storeProps.register = function () {
+      return {
+        messages: {
+          create: this.onCreate;
+        }
+      };
+    };
+```
+Note that `this` in the above example refers to the store being created.
+
+Furthermore, this syntax abstracts away even more complexity by allowing you to register listeners with one or multiple actions and one or multiple categories simultaneously.
+
+```
+    {
+      CATEGORY1: {
+        CATEGORY1_ACTION1: this.ONTHISACTION,
+        CATEGORY1_ACTION2: this.ONANOTHERACTION,
+      },
+      CATEGORY2: {
+        ACTION1: this.ONFINALACTION
+      }
+    }
+```
+
+### <a id="createStore-actionStore"></a>Return - `actionStore` - type: OBJECT [#](#createStore-actionStore)
+Returns an `TuxStore` instance.
+
+## <a id="TuxActionStores-Complete-Example"></a>Tux ActionStores Complete Example [#](#TuxActionStores-Complete-Example)
+
+Taking everything we have learned, let's look at a full example.
+
+```
+    var ActionStores = require('tux/Stores/ActionStores');
+    var Actions = require('tux/Actions');
+
+    var messageActions = Actions.createActionCategory({
+      category: 'messages',
+      source: 'message views',
+      actions: ['create']
+    });
+
+    var messageStoreProperties = {
+      _messages: [],
+
+      onCreate: function (message) {
+        this._messages.push(message);
+        this.emitChange();
+      },
+
+      register: function () {
+        return {
+          messages: {
+            create: this.onCreate;
+          }
+        };
+      }
+    };
+
+    var messageActionStore = ActionStores.createActionStore(messageStoreProperties);
+
+```
```

Comment:
Add summary at the end explaining the example.  Something like:

Hopefully this shows the power of the `ActionStore`. It allows us to easily and semantically describe the action categories and verbs that our store is listening for and the callbacks it will invoke in response to those actions.